File: src/main/java/no/nav/pensjon/regler/domain/kode/KravlinjeTypeCti.java
package no.nav.pensjon.regler.domain.kode;

public class KravlinjeTypeCti extends TypeCti {

    private static final long serialVersionUID = -2393857741445942723L;

    /**
     * Denne er true dersom kravet er en hovedytelse (AP, UP osv).Default false.
     */
    private boolean hovedKravlinje = false;

    /**
     * Copy Constructor
     * 
     * @param kravlinjeTypeCti a <code>KravlinjeTypeCti</code> object
     */
    public KravlinjeTypeCti(KravlinjeTypeCti kravlinjeTypeCti) {
        super(kravlinjeTypeCti);
        hovedKravlinje = kravlinjeTypeCti.hovedKravlinje;
    }

    public KravlinjeTypeCti() {
        super();
    }

    public KravlinjeTypeCti(String kode) {
        super(kode);
    }

    public boolean isHovedKravlinje() {
        return hovedKravlinje;
    }

    public void setHovedKravlinje(boolean hovedKravlinje) {
        this.hovedKravlinje = hovedKravlinje;
    }

}

File: src/main/java/no/nav/pensjon/regler/domain/kode/KravlinjeTypeCti.kt
package no.nav.pensjon.regler.domain.kode

class KravlinjeTypeCti : TypeCti {
    /**
     * Denne er true dersom kravet er en hovedytelse (AP, UP osv).Default false.
     */
    var isHovedKravlinje = false

    constructor(kravlinjeTypeCti: KravlinjeTypeCti) : super(kravlinjeTypeCti) {
        isHovedKravlinje = kravlinjeTypeCti.isHovedKravlinje
    }

    constructor() : super()
    constructor(kode: String?) : super(kode)
}